cpp/src/arrow/CMakeLists.txt

```diff
@@ -52,6 +52,7 @@ if (ARROW_COMPUTE)
   set(ARROW_SRCS ${ARROW_SRCS}
     compute/cast.cc
     compute/context.cc
+    compute/hash_kernels.cc
```

Comment:
In a subsequent PR we should create the `src/arrow/compute/kernels` directory for cleaner organization of code